### PR TITLE
Migrate to Regression Data Completely

### PR DESCRIPTION
**Pull request by atharva-2001**
### :pencil: Description

**Type:** :biohazard: `breaking change` | :vertical_traffic_light: `testing`  | :roller_coaster: `infrastructure`

This PR aims to remove references of refdata from the tardis repo.
https://github.com/tardis-sn/tardis-regression-data/pull/21 and https://github.com/tardis-sn/tardis-regression-data/pull/22 will need to be merged before in order for tests to pass.

### :pushpin: Resources

Examples, notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

File: docs/io/configuration/components/models/data/config_init_trad.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ supernova:
   time_explosion: 16 day
   distance: 8.32 Mpc
 
-# standard atomic data base; get it from the tardis-refdata repository
+# standard atomic data base; get it from the tardis-regression repository
```
Comment (andrewfullard) [inline docs/io/configuration/components/models/data/config_init_trad.yml:12]: Use the full repository name everywhere, `tardis-regression-data`

Comment (atharva-2001) [reply docs/io/configuration/components/models/data/config_init_trad.yml:12]: sure

Comment (atharva-2001) [reply docs/io/configuration/components/models/data/config_init_trad.yml]: Done in [`214fdeb` (#2808)](https://github.com/tardis-sn/tardis/pull/2808/commits/214fdebbd1d92a820c910f5a4b637a4c1f33dca4)

**Review by andrewfullard — COMMENTED** [review]
I'm happy with this, I just want @wkerzendorf to sign off as well on the regression data PRs, then everything here should work as expected.

**Comment by andrewfullard** [comment]
Can you fix the benchmarks?

**Comment by atharva-2001** [comment]
I think it is because of the YAML file not being updated because it is a
pull request target event. The YAML file was picked up from the master
branch instead of this branch.

**Comment by atharva-2001** [comment]
I think it should all work once the PR is merged!

**Review by andrewfullard — APPROVED** [review]

**Review by andrewfullard — APPROVED** [review]

**Review by Vivek-kr22 — COMMENTED** [review]
Key Changes Made:
1. Error Message Update: The error message now directs users to the new regression data repository for atomic data.
2. Placeholder for Parsing Logic: The comment indicates where the existing parsing logic should be implemented, which should be adapted to work with the new regression data format.


def parse_atom_data(config, atom_data=None):
    try:

    except Exception as e:
        print(
            e,
            "Error might be from the use of an old-format of the atomic database, \n"
            "please see https://github.com/tardis-sn/tardis-regression-data/tree/master/atom_data "
            "for the most recent version.",
        )
        raise

**Pull request merged**